src/testing/transaction-logger.ts

```diff
@@ -53,7 +58,22 @@ export class TransactionLogger {
   async waitForIndexer(indexer: Indexer) {
     if (this._sentTransactionIds.length === 0) return
     const lastTxId = this._sentTransactionIds[this._sentTransactionIds.length - 1]
-    await runWhenIndexerCaughtUp(() => indexer.lookupTransactionByID(lastTxId).do())
+    await runWhenIndexerCaughtUp(async () => {
+      try {
+        await indexer.lookupTransactionByID(lastTxId).do()
+      } catch (e) {
+        // If the txid lookup failed, then try to look up the last valid round
+        // If that round exists, then we know indexer is caught up
+        if (this._latestLastValidRound) {
+          await indexer.lookupBlock(this._latestLastValidRound).do()
+          console.debug(
```

Comment:
```suggestion
          Config.getLogger().debug(
```
Will also need the Config import